src/task_monitor/tasks/process_identities.rs

```diff
@@ -394,3 +445,199 @@ async fn commit_identities(
+        merkle_proofs.len(),
+        batch_size,
+        "Mismatch between merkle proofs and batch size."
+    );
+
+    // With the updates applied we can grab the value of the tree's new root and
+    // build our identities for sending to the identity manager.
+    let post_root: U256 = latest_tree_from_updates.root().into();
+    let identity_commitments: Vec<Identity> = commitments
+        .iter()
+        .zip(merkle_proofs)
+        .map(|(id, prf)| {
+            let commitment: U256 = id.into();
+            let proof: Vec<U256> = prf
+                .0
+                .iter()
+                .map(|branch| match branch {
+                    Branch::Left(v) | Branch::Right(v) => U256::from(*v),
+                })
+                .collect();
+            Identity::new(commitment, proof)
+        })
+        .collect();
+
+    identity_manager.validate_merkle_proofs(&identity_commitments)?;
+
+    // We prepare the proof before reserving a slot in the pending identities
+    let proof =
+        IdentityManager::prepare_deletion_proof(prover, pre_root, &identity_commitments, post_root)
+            .await?;
+
+    #[allow(clippy::cast_precision_loss)]
+    PENDING_IDENTITIES_CHANNEL_CAPACITY.observe(pending_batch_submissions_queue.len().await as f64);
+
+    // This queue's capacity provides us with a natural back-pressure mechanism
+    // to ensure that we don't overwhelm the identity manager with too many
+    // identities to mine.
+    let permit = pending_batch_submissions_queue.reserve().await;
+
+    todo!(
+        "TODO, FIXME: do we need start_index for deletions? This value is set for deletions to \
+         compile for now"
+    );
+    let start_index = 0;
+
+    info!(start_index, ?pre_root, ?post_root, "Submitting batch");
+
+    // With all the data prepared we can submit the identities to the on-chain
+    // identity manager and wait for that transaction to be mined.
+    let deletion_indices = updates
+        .iter()
+        .map(|f| f.update.leaf_index as u32)
+        .collect::<Vec<u32>>();
+
+    let transaction_id = identity_manager
+        .delete_identities(proof, pre_root, deletion_indices, post_root)
+        .await
+        .map_err(|e| {
+            error!(?e, "Failed to insert identity to contract.");
+            e
+        })?;
+
+    info!(
+        start_index,
+        ?pre_root,
+        ?post_root,
+        ?transaction_id,
+        "Batch submitted"
+    );
+
+    // The transaction will be awaited on asynchronously
+    permit
+        .send(PendingBatchSubmission {
+            transaction_id,
+            pre_root,
+            post_root,
+            start_index,
+        })
+        .await;
+
+    // Update the batching tree only after submitting the identities to the chain
+    batching_tree.apply_updates_up_to(post_root.into());
+
+    info!(start_index, ?pre_root, ?post_root, "Tree updated");
+
+    // TODO: check if any deleted commitments correspond with entries in the
```

Comment:
This part should happen in `mine_identities`

If we do it here we could have an overlap of how long it takes to mine a transaction on chain when both the deleted identity and it's replacement is usable.

Not sure what we should do in case of a reorg though.